airflow-core/docs/tutorial/objectstorage.rst

```diff
@@ -18,117 +18,151 @@
-to register the backend with Duck DB.
+Analyzing the Data with DuckDB
+------------------------------
 
-In Duck DB we can then create a table from the data and run a query on it. The
-query is returned as a dataframe, which could be used for further analysis or
-saved to object storage.
+Now let's analyze that data using SQL with DuckDB.
 
 .. exampleinclude:: /../src/airflow/example_dags/tutorial_objectstorage.py
     :language: python
     :start-after: [START analyze]
     :end-before: [END analyze]
 
-You might note that the ``analyze`` function does not know the original
-path to the object, but that it is passed in as a parameter and obtained
-through XCom. You do not need to re-instantiate the Path object. Also
-the connection details are handled transparently.
+|
 
-Putting it all together
------------------------
+A few key things to note:
 
-The final DAG looks like this, which wraps things so that we can run it:
+- DuckDB supports reading Parquet natively
+- DuckDB and ObjectStoragePath both rely on ``fsspec``, which makes it easy to register the object storage backend
+- We use ``path.fs`` to grab the right filesystem object and register it with DuckDB
+- Finally, we query the Parquet file using SQL and return a pandas DataFrame
+
+Notice that the function doesn't recreate the path manually -- it gets the full path from the previous task using Xcom.
+This makes the task portable and decoupled from earlier logic.
+
+Bringing It All Together
+------------------------
+
+Here's the full DAG that ties everything together:
 
 .. exampleinclude:: /../src/airflow/example_dags/tutorial_objectstorage.py
     :language: python
     :start-after: [START tutorial]
     :end-before: [END tutorial]
+
+|
+
+You can trigger this DAG and view it in the Graph View in the Airflow UI. Each task logs its inputs and outputs clearly,
+and you can inspect returned paths in the Xcom tab.
+
+What to Explore Next
+--------------------
+
+Here are some ways to take this further:
+
+- Use object sensors (like ``S3KeySensor``) to wait for files uploaded by external systems
+- Orchestrate S3-to-GCS transfers or cross-region data syncs
+- Add branching logic to handle missing or malformed files
+- Experiment with different formats like CSV or JSON
+
+**See Also**
+
+- Learn how to securely access cloud services by configuring Airflow connections in the `Managing Connections guide <https://airflow.apache.org/docs/apache-airflow/stable/authoring-and-scheduling/connections.html>`__
+- Build event-driven pipelines that respond to file uploads or external triggers using the `Event-Driven Scheduling framework <https://airflow.apache.org/docs/apache-airflow/stable/authoring-and-scheduling/event-scheduling.html>`__
+- Reinforce your understanding of decorators, return values, and task chaining with the `TaskFlow API guide <https://airflow.apache.org/docs/apache-airflow/stable/core-concepts/taskflow.html>`__
```

Comment:
We should have relative links here or reference to the section.

```suggestion
- Learn how to securely access cloud services by configuring Airflow connections in the :doc:`Managing Connections guide </authoring-and-scheduling/connections>`__
- Build event-driven pipelines that respond to file uploads or external triggers using the :doc:`Event-Driven Scheduling framework </authoring-and-scheduling/event-scheduling>`__
- Reinforce your understanding of decorators, return values, and task chaining with the :doc:`TaskFlow API guide </core-concepts/taskflow.html>`__
```

Something like this should work

Reply:
Squashed and fixed in https://github.com/apache/airflow/pull/49170/commits/6375662e550709a4456fac902013be3eb24a1e78